### PR TITLE
add signal psrpc service

### PR DESCRIPTION
**Pull request by paulwe**

### CODE REVIEW

**Comment by paulwe** [comment]
we have to wait for whatever upstream issues are going on with staticcheck for the tests to run but they pass locally

File: pkg/service/signal.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	r.server.Kill()
+}
+
+type signalService struct {
+	region         string
+	sessionHandler SessionHandler
+}
+
+func (r *signalService) RelaySignal(stream psrpc.ServerStream[*rpc.RelaySignalResponse, *rpc.RelaySignalRequest]) (err error) {
+	ctx, cancel := context.WithCancel(psrpc.NewContextWithIncomingHeader(context.Background(), psrpc.IncomingHeader(stream.Context())))
```
Comment (davidzhao) [inline pkg/service/signal.go:79]: what would be in the header / context? a bit of comments 🙏

**Review by davidzhao — APPROVED** [review]
Nice work! just a couple of nits.

File: pkg/service/signal.go (hunk 1 above)
Comment (paulwe) [reply pkg/service/signal.go:79]: this is for smuggling the auth info to the server for cloud - do we want to call ths out in oss?

Comment (davidzhao) [reply pkg/service/signal.go:79]: I think would be helpful to note that it's to pass along context such as auth info to the other instance.

**Pull request merged**